src/App.js

```diff
@@ -14,11 +15,93 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends Component {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  goodsSelector = (goods) => {
+    switch (goods.length) {
+      case 0:
+        return 'No goods selected';
+      case 1:
+        return `${goods[0]} is selected`;
+      default:
+        return `${goods.slice(0, goods.length - 1).join(', ')}`
+        + ` and ${goods[goods.length - 1]} are selected`;
+    }
+  }
+
+  removeGood = (good) => {
+    this.setState(state => ({
+      selectedGoods: state.selectedGoods.filter(product => product !== good),
+    }));
+  };
+
+  addGood = (good) => {
+    this.setState(state => ({
+      selectedGoods: [...state.selectedGoods, good],
+    }));
+  };
+
+  clearAllGoods = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <div className="App__wrapper">
+          <h1 className="App__header">
+            {this.goodsSelector(selectedGoods)}
+          </h1>
+          <button
+            type="button"
+            className={classNames('App__clear-button', {
+              'App__clear-button--disable': selectedGoods.length === 0,
+            })}
+            onClick={this.clearAllGoods}
+          >
+            ✗
+          </button>
+        </div>
+        <ul>
+          {goodsFromServer.map(good => (
+            <li
+              className={
+                selectedGoods.includes(good)
+                  ? 'App__good App__good--active'
+                  : 'App__good'
+              }
+              key={good}
+            >
+              <span>{good}</span>
+              <button
+                className="App__addButton"
+                type="button"
+                onClick={() => (
+                  !selectedGoods.includes(good) && this.addGood(good)
+                )}
+              >
+                Add
+              </button>
+              <button
+                className="App__removeButton"
+                type="button"
+                onClick={() => (
+                  selectedGoods.includes(good) && this.removeGood(good)
+                )}
+              >
+                Remove
+              </button>
```

Comment:
now you have both add and remove buttons and in my previous review I meant to render one of the button on condition
`selectedGoods.includes(good) ? <button>remove</button> : <button>add</button>`